### PR TITLE
Chore: Upgrade core to RN 0.75

### PR DESCRIPTION
**Pull request by antonis**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

**Based on: https://github.com/getsentry/sentry-react-native/pull/4155**

- Upgrades packages/core to RN 0.75
- [Fixes prerelease property type](https://github.com/getsentry/sentry-react-native/pull/4173/commits/c32f8cbf1486c7ca7b019c321f89e4cc0847bbee)
- Sets the `REACT_NATIVE_PATH` that was causing a failure in the `RNSentryCocoaTester` due to [a newly introduced check in 0.75.4](https://github.com/facebook/react-native/blame/v0.74.5/packages/react-native/ReactCommon/React-Fabric.podspec#L304)

## :bulb: Motivation and Context

See https://github.com/getsentry/sentry-react-native/pull/4155#issuecomment-2411583543

## :green_heart: How did you test it?
CI and manual test

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

#skip-changelog

### CODE REVIEW

**Review by lucas-zimerman — APPROVED** [review]
Looks good to me, LGTM!

File: packages/core/babel.config.js [hunk 1]
```diff
@@ -1,3 +1,9 @@
 module.exports = {
   presets: ['module:metro-react-native-babel-preset'],
+  plugins: [
+    ['@babel/plugin-transform-class-properties', { loose: true }],
+    ['@babel/plugin-transform-private-methods', { loose: true }],
+    ['@babel/plugin-transform-private-property-in-object', { loose: true }]
+  ],
```
Comment (krystofwoldrich) [inline packages/core/babel.config.js:7]: Why are these plugins needed?

Comment (antonis) [reply packages/core/babel.config.js:7]: I added those to fix test failures in `test/tracing/reactnavigation.ttid.test.tsx`
```
Class private methods are not enabled. Please add `@babel/plugin-transform-private-methods` to your configuration.
```

**Review by krystofwoldrich — APPROVED** [review]
Thank you! Looks good! 🚀

**Pull request merged**